common/modules/buffer.jsm

```diff
@@ -731,9 +731,14 @@ var Buffer = Module("Buffer", {
         }
 
         let link = DOM("link[href][rev=canonical], \
+                        link[href][rel=shorturl], \
                         link[href][rel=shortlink]", doc)
                        .filter(sane)
                        .attr("href");
+
+        if (!link)
+            link = DOM("link[href][rel=canonical]", doc).filter(sane).attr("href");
```

Comment:
`rel=canonical` is a hint for search engines as to the canonical URL of content. It usually won't contain a shortened version of the URL, so it's not appropriate here.